src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import {
+  TransitionGroup,
+  CSSTransition,
+} from 'react-transition-group';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { StatusEnum } from '../../types/StatusEnum';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  todosIdToDelete: number[];
+  todosIdToUpdate: number[];
+  onDeleteTodo: (todoId: number) => void;
+  onChangeTitle: (todoId: number, newTitle: string) => void;
+  onChangeCompletedStatus: (todoId: number, isCompleted: boolean) => void;
+  filter: StatusEnum;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  filter,
+  onChangeCompletedStatus,
+  onChangeTitle,
+  onDeleteTodo,
+  tempTodo,
+  todosIdToDelete,
+  todosIdToUpdate,
+}) => {
+  return (
+    <section data-cy="TodoList" className="todoapp__main">
+      <TransitionGroup>
+        {todos.filter(todo => {
+          switch (filter) {
+            case StatusEnum.Active:
+              return !todo.completed;
+            case StatusEnum.Completed:
+              return todo.completed;
+            case StatusEnum.All:
+            default:
+              return todo;
+          }
+        }).map(todo => (
+          <CSSTransition
+            key={todo.id}
+            timeout={300}
+            classNames="item"
+          >
+            <TodoItem
+              todo={todo}
+              todosIdToDelete={todosIdToDelete}
+              todosIdToUpdate={todosIdToUpdate}
+              onDeleteTodo={onDeleteTodo}
+              onChangeTitle={onChangeTitle}
+              onChangeCompletedStatus={onChangeCompletedStatus}
+            />
+          </CSSTransition>
+        ))}
+        {tempTodo && (
+          <CSSTransition
+            key={0}
+            timeout={300}
+            classNames="item"
+          >
+            <TodoItem
+              todo={tempTodo}
+              todosIdToDelete={todosIdToDelete}
+              todosIdToUpdate={todosIdToUpdate}
+              onDeleteTodo={onDeleteTodo}
+              onChangeTitle={onChangeTitle}
+              onChangeCompletedStatus={onChangeCompletedStatus}
+            />
+          </CSSTransition>
```

Comment:
You repeat this twice, maybe it's better to keep with `TodoItem`, but I'm not sure if `CSSTransition` works properly in this case.

Also, a few hints:
- to avoid props drilling you can use context.
- if you don't need at this level props and just want to pass them to the next component you can use the `spread` syntax. Example:
```javascript
function Profile(props) {
  return (
    <div className="card">
      <Avatar {...props} />
    </div>
  );
}
```
https://react.dev/learn/passing-props-to-a-component
- also, in this case, you can reuse props type